noether_gui/src/widgets/tpp_widget.cpp

```diff
@@ -0,0 +1,71 @@
+#include <noether_gui/widgets/tpp_widget.h>
+#include "ui_tpp_widget.h"
+#include <noether_gui/widgets/tpp_pipeline_widget.h>
+#include <noether_gui/utils.h>
+
+#include <pcl/io/vtk_lib_io.h>
+#include <QFileDialog>
+#include <QMessageBox>
+#include <QStandardPaths>
+#include <QTextStream>
+
+namespace noether
+{
+TPPWidget::TPPWidget(boost_plugin_loader::PluginLoader loader, QWidget* parent) : QWidget(parent), ui_(new Ui::TPP())
+{
+  ui_->setupUi(this);
+
+  auto pipeline = new TPPPipelineWidget(std::move(loader), this);
+  ui_->scroll_area->setWidget(pipeline);
+
+  connect(ui_->push_button_load_mesh, &QPushButton::clicked, this, &TPPWidget::onLoadMesh);
+  connect(ui_->push_button_plan, &QPushButton::clicked, this, &TPPWidget::onPlan);
+}
+
+std::vector<ToolPaths> TPPWidget::getToolPaths() { return tool_paths_; }
+
+void TPPWidget::onLoadMesh(const bool)
+{
+  const QString home = QStandardPaths::standardLocations(QStandardPaths::HomeLocation).at(0);
+  const QString file = QFileDialog::getOpenFileName(this, "Load mesh file", home, "Mesh files (*.ply *.stl)");
+  ui_->line_edit_mesh->setText(file);
+}
+
+void TPPWidget::onPlan(const bool)
```

Comment:
Is the anonymous `const bool` just here to match the required signal/slot prototype for a button?  If not, could it just be a no-arg function?

Reply:
Yes, the slot requires a bool to indicate whether the push button is toggled or not which is irrelevant for this use-case